backend/app/app/api/api_v1/endpoints/monitoringevent.py

```diff
@@ -1,179 +1,338 @@
-            else:
-                json_compatible_item_data["locationInfo"] = {'cellId' : UE.Cell.cell_id, 'gNBId' : UE.Cell.gNB.gNB_id}
-        else:
-            json_compatible_item_data["locationInfo"] = {'cellId' : None, 'gNBId' : None}
-
-        http_response = JSONResponse(content=json_compatible_item_data, status_code=200)
-        add_notifications(http_request, http_response, False)
-        
-        return http_response 
-    #Subscription
-    elif item_in.monitoringType == "LOCATION_REPORTING" and item_in.maximumNumberOfReports>1:
-        
-        #Check if subscription with externalid exists
-        if crud_mongo.read_by_multiple_pairs(db_mongo, db_collection, externalId = item_in.externalId, monitoringType = item_in.monitoringType):
-            raise HTTPException(status_code=409, detail=f"There is already an active subscription for UE with external id {item_in.externalId} - Monitoring Type = {item_in.monitoringType}")
-   
-        json_data = jsonable_encoder(item_in.dict(exclude_unset=True))
-        json_data.update({'owner_id' : current_user.id, "ipv4Addr" : UE.ip_address_v4})
-
-        inserted_doc = crud_mongo.create(db_mongo, db_collection, json_data) 
-        
-        #Create the reference resource and location header
-        link = str(http_request.url) + '/' + str(inserted_doc.inserted_id)
-        response_header = {"location" : link}
-        
-        #Update the subscription with the new resource (link) and return the response (+response header)
-        crud_mongo.update_new_field(db_mongo, db_collection, inserted_doc.inserted_id, {"link" : link})
-
-        #Retrieve the updated document | UpdateResult is not a dict
-        updated_doc = crud_mongo.read_uuid(db_mongo, db_collection, inserted_doc.inserted_id)
-        updated_doc.pop("owner_id")
+    if item_in.self is not None:
+        raise HTTPException(
+            status_code=400, detail="The self attribute must not be set"
+        )
 
+    db_mongo = client.fastapi
 
-        http_response = JSONResponse(content=updated_doc, status_code=201, headers=response_header)
-        add_notifications(http_request, http_response, False)
-        
-        return http_response
-    elif (item_in.monitoringType == "LOSS_OF_CONNECTIVITY" or item_in.monitoringType == "UE_REACHABILITY") and item_in.maximumNumberOfReports == 1:
-        return JSONResponse(content=jsonable_encoder(
-            {
-                "title" : "The requested parameters are out of range",
-                "invalidParams" : {
-                    "param" : "maximumNumberOfReports",
-                    "reason" : "\"maximumNumberOfReports\" should be greater than 1 in case of LOSS_OF_CONNECTIVITY event"
+    ue = None
+
+    id = ObjectId()
+    item_in.self = parse_obj_as(Link, f"{http_request.url}/{id}")
+
+    if item_in.ipv4Addr is not None:
+        id_value = str(item_in.ueIpv4Addr)
```

Comment:
```suggestion
        id_value = str(item_in.ipv4Addr)
```
This property doesn't exist